### PR TITLE
Feature/include all level diffing

### PR DESCRIPTION
**Pull request by kandelvijaya**
## breaking changes
- `children` -> `innerDiffableItems`. We found most grap/tree structure have internal property called `children` which would conflict with requirement getter. This was utterly confusing when 

```swift 
struct Component: Diffable {
     let name: String
     let children: [Component]?       // Swift would assume this to fulfill the requirement which caused confusion
}
```

## New addtions
- `diffAllLevels` is included in the library. It seems to be useful in couple of repeated use cases. 
- `diffHash` and `innerDiffableItems` are better described in both `README.md` and header documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,34 +79,99 @@ let orderedChangeSet = orderedOperation(from: changeSet)
 
 ### Advanced usage & notes
 1. This algorithm works accurately with value types `Struct`'s. Please refrain from using reference type (`Class` instance). When you must use class instance / object, you might get more updates than you expect. If you want to resolve this issue for your use case please DM me www.twitter.com/kandelvijaya
-2. Tree diffing is possible. However not something the library encourages due to added complexity O(n^2). If you so choose to diff then please use this routine:
```
Comment (kandelvijaya) [inline README.md:82]: moved to library.

File: Sources/FastDiff/Diffable.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
+//
```
Comment (kandelvijaya) [inline Sources/FastDiff/Diffable.swift:1]: Moved from directory. As is.

File: README.md [hunk 3]
```diff
@@ -79,34 +79,99 @@ let orderedChangeSet = orderedOperation(from: changeSet)
...
-   ```
+2. Tree diffing is possible. However not something the library encourages due to added complexity O(n^2). If you so choose to diff then please use `diffAllLevel(,)`
 3. The complexity of Graph diffing depends on graph structure. For Trees, its O(n^2). Please note that this change set is not mergeable to the original tree. To circumvent this limitation, use a node with indexes or indepath that points to the graph position implicitly. 
 
 ### Concept and advanced usage in List View Controller (iOS)
 Please check out this presentation slides that I gave at [@mobiconf 2018](https://drive.google.com/file/d/1eY0k_5sHBDgK6Qx6-VR3HTmCQEi9qaW3/view?usp=sharing)
+  
+
+
+## Why is nested diffing important? Tutorial/HowTo
```
Comment (kandelvijaya) [inline README.md:90]: Better explanation of why nested diff is imporntant and how library expects to be done.

File: Sources/FastDiff/InternalDiff.swift [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    return accumulator
+}
+
+
+/// Calculates diff from entire graph going deeper as it finds `update` on container.
+/// It is greedy algorithm.
+/// - NOTE: Profile when running on main thread.
+///
+/// - Complexity:- O(allEdges * nlog(n))
+public func diffAllLevel<T>(_ oldContent: [T], _ newContent: [T]) -> [DiffOperation<T>] where T: Diffable, T.InternalItemType == T {
```
Comment (kandelvijaya) [inline Sources/FastDiff/InternalDiff.swift:34]: New addition.

File: Tests/FastDiffTests/XCTestManifests.swift [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import XCTest
 
-#if !os(macOS)
+#if !canImport(ObjectiveC)
```
Comment (kandelvijaya) [inline Tests/FastDiffTests/XCTestManifests.swift:3]: this changed.

File: Sources/FastDiff/InternalDiff.swift (hunk 4 above)
Comment (osama10) [reply Sources/FastDiff/InternalDiff.swift:34]: tests?

**Pull request merged**